lib/dry/schema/dsl.rb

```diff
@@ -67,7 +67,7 @@ class DSL
       option :parent, Types::Coercible::Array, default: -> { EMPTY_ARRAY.dup }, as: :parents
 
       # @return [Config] Configuration object exposed via `#configure` method
-      option :config, optional: true, default: proc { parent ? parent.config.dup : Config.new }
+      option :config, optional: true, default: proc { (parent || Schema).config.dup }
```

Comment:
Would it be possible to encapsulate this in the parent, so that this line could remain unchanged?

Reply:
I don't think it's possible, since this line checks for the presence of the parent to determine whether to use the parent's config or the global config.